sale_product_set/models/res_partner.py

```diff
@@ -0,0 +1,20 @@
+from odoo import models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def write(self, vals):
+        res = super().write(vals)
+        ir_config_param = self.env["ir.config_parameter"].sudo()
+
+        if "active" in vals and ir_config_param.get_param(
+            "sale.archive_partner_product_sets"
+        ):
```

Comment:
```suggestion
        if "active" in vals and str2bool(
            ir_config_param.get_param("sale.archive_partner_product_sets")
        ):
```
Use `str2bool` otherwise any value would be evaluated to True if the key exists